validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -493,9 +502,14 @@ private byte[] getFileBytes(String dataFile) {
   }
 
   private String deviceConfigString() {
+    if (configJson == null) {
+      return null;
+    }
+
     try {
-      JsonNode configJson = OBJECT_MAPPER_STRICT.valueToTree(deviceConfigObject());
-      new MessageDowngrader("config", configJson).downgrade(baseVersion);
+      if (config.canBeDowngraded()) {
```

Comment:
I'd phrase this as "shouldBeDowngraded" -- I think it could (can) be downgraded, it's just that we may not want to.

Reply:
Changed